diff --git a/arch/arm/lib/board.c b/arch/arm/lib/board.c
index 5d8116b..1e1cf81 100644
--- a/arch/arm/lib/board.c
+++ b/arch/arm/lib/board.c
@@ -104,23 +104,23 @@ extern int check_ddr_training(void);
  ************************************************************************
  * May be supplied by boards if desired
  */
-void inline __coloured_LED_init (void) {}
+void __coloured_LED_init (void) {}
 void coloured_LED_init (void) __attribute__((weak, alias("__coloured_LED_init")));
-void inline __red_LED_on (void) {}
+void __red_LED_on (void) {}
 void red_LED_on (void) __attribute__((weak, alias("__red_LED_on")));
-void inline __red_LED_off(void) {}
+void __red_LED_off(void) {}
 void red_LED_off(void) __attribute__((weak, alias("__red_LED_off")));
-void inline __green_LED_on(void) {}
+void __green_LED_on(void) {}
 void green_LED_on(void) __attribute__((weak, alias("__green_LED_on")));
-void inline __green_LED_off(void) {}
+void __green_LED_off(void) {}
 void green_LED_off(void) __attribute__((weak, alias("__green_LED_off")));
-void inline __yellow_LED_on(void) {}
+void __yellow_LED_on(void) {}
 void yellow_LED_on(void) __attribute__((weak, alias("__yellow_LED_on")));
-void inline __yellow_LED_off(void) {}
+void __yellow_LED_off(void) {}
 void yellow_LED_off(void) __attribute__((weak, alias("__yellow_LED_off")));
-void inline __blue_LED_on(void) {}
+void __blue_LED_on(void) {}
 void blue_LED_on(void) __attribute__((weak, alias("__blue_LED_on")));
-void inline __blue_LED_off(void) {}
+void __blue_LED_off(void) {}
 void blue_LED_off(void) __attribute__((weak, alias("__blue_LED_off")));
 
 /************************************************************************
diff --git a/common/Makefile b/common/Makefile
index 580bb3a..433f033 100644
--- a/common/Makefile
+++ b/common/Makefile
@@ -341,11 +341,9 @@ endif #/*end CONFIG_HIFMC_NAND*/
 endif #/*end CONFIG_HI3556*/
 endif #/*end arm-hisiv600-linux-*/
 
-ifeq ($(CROSS_COMPILE),arm-hisiv510-linux-)
 ifdef CONFIG_HI3518EV200
 	cp cmd_bootss2_v510_hi3518ev200 bootss2.a -rf
 endif
-endif
 
 ifeq ($(CROSS_COMPILE),arm-hisiv610-linux-)
 ifdef CONFIG_HI3518EV200
diff --git a/common/main.c b/common/main.c
index 933864e..3fbaf57 100644
--- a/common/main.c
+++ b/common/main.c
@@ -48,7 +48,7 @@ DECLARE_GLOBAL_DATA_PTR;
 /*
  * Board-specific Platform code can reimplement show_boot_progress () if needed
  */
-void inline __show_boot_progress (int val) {}
+void __show_boot_progress (int val) {}
 void show_boot_progress (int val) __attribute__((weak, alias("__show_boot_progress")));
 
 #if defined(CONFIG_BOOT_RETRY_TIME) && defined(CONFIG_RESET_TO_RETRY)
diff --git a/lib/time.c b/lib/time.c
index a309c26..60ea897 100644
--- a/lib/time.c
+++ b/lib/time.c
@@ -41,3 +41,9 @@ void udelay(unsigned long usec)
 		usec -= kv;
 	} while(usec);
 }
+
+void wait_ms(unsigned long msec)
+{
+	while (msec--)
+		udelay(1000);
+}
